python_ta/debug/snapshot.py

```diff
@@ -56,6 +58,7 @@ def get_filtered_local_variables(
 
 
 def snapshot(
+    id_tracker: IDTracker,
```

Comment:
Generally speaking, adding parameters to existing functions should be done quite carefully to preserve backwards compatibility.

In this case, (1) `id_tracker` should be added to the end of the list, and (2) it should be optional (default value `None`), as the only time it's needed is when the user wants to track ids across multiple calls to `snapshot`. This is only really necessary for the `SnapshotTracer` class, but not other uses of `snapshot`. Same applies to `snapshot_to_json`.

Also make sure to add documentation for this attribute.